builtin/logical/transit/stepwise_test.go

```diff
@@ -0,0 +1,231 @@
+			}
+			if len(d.Keys) > 0 && d.Keys[0] != name {
+				return fmt.Errorf("bad name: %#v", d)
+			}
+			if len(d.Keys) != 1 {
+				return fmt.Errorf("only 1 key expected, %d returned", len(d.Keys))
+			}
+			return nil
+		},
+	}
+}
+
+func testAccStepwiseReadPolicy(t *testing.T, name string, expectNone, derived bool) stepwise.Step {
+	t.Helper()
+	return testAccStepwiseReadPolicyWithVersions(t, name, expectNone, derived, 1, 0)
+}
+
+func testAccStepwiseReadPolicyWithVersions(t *testing.T, name string, expectNone, derived bool, minDecryptionVersion int, minEncryptionVersion int) stepwise.Step {
+	t.Helper()
+	return stepwise.Step{
+		Operation: stepwise.ReadOperation,
+		Path:      "keys/" + name,
+		Assert: func(resp *api.Secret, err error) error {
+			t.Helper()
+			if resp == nil && !expectNone {
+				return fmt.Errorf("missing response")
+			} else if expectNone {
+				if resp != nil {
+					return fmt.Errorf("response when expecting none")
+				}
+				return nil
+			}
+			var d struct {
+				Name                 string           `mapstructure:"name"`
+				Key                  []byte           `mapstructure:"key"`
+				Keys                 map[string]int64 `mapstructure:"keys"`
+				Type                 string           `mapstructure:"type"`
+				Derived              bool             `mapstructure:"derived"`
+				KDF                  string           `mapstructure:"kdf"`
+				DeletionAllowed      bool             `mapstructure:"deletion_allowed"`
+				ConvergentEncryption bool             `mapstructure:"convergent_encryption"`
+				MinDecryptionVersion int              `mapstructure:"min_decryption_version"`
+				MinEncryptionVersion int              `mapstructure:"min_encryption_version"`
+			}
+			if err := mapstructure.Decode(resp.Data, &d); err != nil {
+				return err
+			}
+
+			if d.Name != name {
+				return fmt.Errorf("bad name: %#v", d)
+			}
+			if os.Getenv("TRANSIT_ACC_KEY_TYPE") == "CHACHA" {
+				if d.Type != keysutil.KeyType(keysutil.KeyType_ChaCha20_Poly1305).String() {
+					return fmt.Errorf("bad key type: %#v", d)
+				}
+			} else if d.Type != keysutil.KeyType(keysutil.KeyType_AES256_GCM96).String() {
+				return fmt.Errorf("bad key type: %#v", d)
+			}
+			// Should NOT get a key back
+			if d.Key != nil {
+				return fmt.Errorf("bad: %#v", d)
+			}
+			if d.Keys == nil {
+				return fmt.Errorf("bad: %#v", d)
+			}
+			if d.MinDecryptionVersion != minDecryptionVersion {
+				return fmt.Errorf("bad: %#v", d)
+			}
+			if d.MinEncryptionVersion != minEncryptionVersion {
+				return fmt.Errorf("bad: %#v", d)
+			}
+			if d.DeletionAllowed == true {
+				return fmt.Errorf("bad: %#v", d)
+			}
+			if d.Derived != derived {
+				return fmt.Errorf("bad: %#v", d)
+			}
+			if derived && d.KDF != "hkdf_sha256" {
+				return fmt.Errorf("bad: %#v", d)
+			}
+			return nil
+		},
+	}
+}
+
+func testAccStepwiseEncryptContext(
+	t *testing.T, name, plaintext, context string, decryptData map[string]interface{}) stepwise.Step {
```

Comment:
I don't have a concrete proposal as to how we can pass data between steps yet, but it feels a bit weird to be mutating our input data as opposed to returning a copy of the response's data explicitly.

Reply:
Deferring and tracking this internally, to follow up in the coming weeks